### DISCUSSION TITLE
What is the way to organize a multi-register render?

### DISCUSSION THREAD

**Discussion by SergiiSharpov** [skypjack/entt#831] [General]
I'm new to your awesome library and was rereading the documentation as much as possible, but still have a few questions and one of them is based on the following functionality:

I have the next independent scenes/features (call it what you like):
1. Main game
2. User Interface
3. Menus on request (Menu on Esc, Inventory menu, hundreds of other menus, its just for example)

So what way should I go using an Entt, to have the ability to render all of the parts independently?
In my head, it looks like:
```
renderer->render(game, camera);
renderer->clearDepth();
renderer->render(engineUI, uiCamera);
if (someWindow->isActive) {
  renderer->clearDepth();
  renderer->render(someWindow, uiCamera);
}
```

Before reading the documentation, I was thinking about groups, but as I understood correctly after reading, they are just for grouping the components during the iterations over the registry.

Should each scene have its own registry in this case?
Will there be some conflicts in the internal Entt system if I'm gonna use a registry for each scene?

The main idea is to have an optimized sorting/rendering/updating/etc. for each system.
Because as for the UI there might be a lot of computation about how to shift/scale the next UI element based on the previous one.
And a lot of other computations to make a UI as faster as possible, but I do not need to update it each frame, only on request.

Shortly, I want to render specific systems independently without filtering all the entities that are placed in the single registry frame by frame.
What is the way to organize their entities through the Entt?

**Comment by skypjack** [comment]
EnTT is designed as a container (or a bunch of containers if you use storage classes directly and get rid of the registry).
There is no problem in using multiple registries with multiple scenes. All of them will be independent of each other, as well as their entities and their components.

**Reply by SergiiSharpov** [reply]
Thank you!
